### ISSUE TITLE
Rule for initialisms within FROM clauses.

### ISSUE THREAD

**Issue by alanmcruickshank** [sqlfluff/sqlfluff#479]
This is prompted by the excellent work of @piotrgredowski on #473 for #469.

The fishtown sql guide says:

> Avoid table aliases in join conditions (especially initialisms) – it's harder to understand what the table called "c" is compared to "customers".

Which we have implemented a rule for which corrects:

```sql
SELECT u.id, c.first_name, c.last_name, COUNT(o.user_id)
FROM users as u
JOIN customers as c on u.id = c.user_id
JOIN orders as o on u.id = o.user_id
```
to
```sql
SELECT u.id, customers.first_name, customers.last_name, COUNT(orders.user_id)
FROM users as u
JOIN customers on u.id = customers.user_id
JOIN orders on u.id = orders.user_id
```

This is definitely in line with the _letter of the law_. What I'm wondering is whether initialisms in a FROM clause are also part of the same antipattern 🤔 . Should there be an additional rule to avoid initialisms in the `FROM` clause such that the full corrected query is:
```sql
SELECT users.id, customers.first_name, customers.last_name, COUNT(orders.user_id)
FROM users
JOIN customers on users.id = customers.user_id
JOIN orders on users.id = orders.user_id
```

I'm going to ask the fishtown guys what they think, but I'm curious for what other people think.

[I also think that @NiallRees will be amused by this suggestion because he's seen a load of my production sql and knows that this is one of my bad habits]

**Comment by alanmcruickshank** [comment]
We may get feedback for this on the dbt slack channel where I've posted a link to this question. I'm posting a link here in case anyone wants to follow the conversation there: https://getdbt.slack.com/archives/CV2KM39KR/p1602348700008400

**Comment by NiallRees** [comment]
I couldn't possibly comment 😛

Good idea, let's definitely get this in. We'll just have to be mindful of the logic which allows aliases for self-joins. What do you think @piotrgredowski? Fancy doing this one too?

**Comment by piotrgredowski** [comment]
@NiallRees of course I can do this :) to be honest - it was working that way (this rule was reporting issue for initialism for table from `FROM` clause too). I've just added condition to avoid including that table's alias in lint results. Why? Because I read issue name again and realised that it's saying that only aliases for `JOIN` tables should be reported. ;)
So it should be super easy to implement. I can do it if you are sure that you want that.

**Comment by emilieschario** [comment]
I'm pretty confident that the answer here is aliases shouldn't be used for the same reason as you don't want it in join clauses.

**Comment by NiallRees** [comment]
@piotrgredowski ha sorry about that, I copied the issue spec directly from the dbt coding conventions. Thanks for weighing in @emilieschario, it sounds like we're all agreed that it would make sense for this rule to apply to `FROM` clauses as well as joins. Go ahead @piotrgredowski!

**Comment by piotrgredowski** [comment]
@NiallRees so my last question for now is: should it be included in the same rule or should I create new rule to cover only aliases in 'FROM' clause?

**Comment by NiallRees** [comment]
@piotrgredowski good question. Let's keep it in the same rule, and if there is demand to distinguish between `from` and `join` aliasing we can add a configuration parameter later.

**Comment by piotrgredowski** [comment]
I agree that motivation is the same - so I'll just modify existing rule and reflect this modification in rule documentation.

**Comment by NiallRees** [comment]
> I agree that motivation is the same - so I'll just modify existing rule and reflect this modification in rule documentation.
> Edit: @NiallRees can you add hacktoberfest label to issue? 😃

From what I've read I'm meant to add it to the pull request now but I'll add to both to make sure! Thanks!
